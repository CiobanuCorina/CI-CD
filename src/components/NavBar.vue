<template>
  <v-app-bar
      app
      color="white"
      flat
  >
    <v-container class="py-0 fill-height">
      <v-col
          class="text-h4"
          color="grey darken-1"
          size="60"
      >Simple Store</v-col>

      <v-btn
          v-for="link in links"
          :key="link"
          text
      >
        {{ link }}
      </v-btn>

      <v-spacer></v-spacer>

      <v-responsive max-width="260">
        <v-text-field
            dense
            flat
            hide-details
            rounded
            solo-inverted
        ><template v-slot:label>
          Search for products <v-icon style="vertical-align: middle" size="20">
          fas fa-search
        </v-icon>
        </template></v-text-field>
      </v-responsive>
      <v-spacer></v-spacer>
      <v-icon>
        fas fa-shopping-cart fa-2x
      </v-icon>
    </v-container>
  </v-app-bar>
</template>

<script>
export default {
  name: "NavBar",
  data: () => ({
    links: [
      'Home',
      'Info',
      'Contacts',
    ],
  }),
}
</script>

<style scoped>

</style>