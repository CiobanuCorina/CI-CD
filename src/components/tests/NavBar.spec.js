import {createLocalVue, mount} from "@vue/test-utils";
import NavBar from "../NavBar";
import Vuex from "vuex";
import Vuetify from 'vuetify';
import VueRouter from "vue-router";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);
localVue.use(VueRouter);
// localVue.component('VIcon', {
//
// })

const store = new Vuex.Store({
    getters: {
        'products/getSearchSuggestions': jest.fn(),
        'products/getIsSearchLoading': jest.fn()
    }
})

describe('NavBar', () => {
    it('should match snapshot', () => {
        const component = mount(NavBar, {
            store,
            localVue,
            vuetify: new Vuetify({
                icons: {
                    iconfont: 'fa',
                },
            })
        })
        expect(component.element).toMatchSnapshot();
    })
})