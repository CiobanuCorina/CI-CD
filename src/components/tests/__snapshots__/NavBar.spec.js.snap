// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot 1`] = `
<div>
  <v-app-bar-stub
    app="true"
    class="navbar"
    extensionheight="48"
    flat="true"
    src=""
    tag="header"
    value="true"
  >
    <v-container-stub
      class="py-0 fill-height"
      tag="div"
    >
      <v-toolbar-items-stub
        class="hidden-lg-and-up"
      >
        <v-app-bar-nav-icon-stub />
      </v-toolbar-items-stub>
       
      <v-col-stub
        tag="div"
      >
        <v-btn-stub
          activeclass=""
          class="text-h5 hidden-sm-and-down"
          color="grey darken-1"
          tag="button"
          text="true"
          to="/"
          type="button"
        >
          
          Simple Store
        
        </v-btn-stub>
      </v-col-stub>
       
      <v-btn-stub
        activeclass=""
        class="hidden-md-and-down"
        tag="button"
        text="true"
        to="/"
        type="button"
      >
        
        Home
      
      </v-btn-stub>
      <v-btn-stub
        activeclass=""
        class="hidden-md-and-down"
        tag="button"
        text="true"
        to="/info"
        type="button"
      >
        
        Info
      
      </v-btn-stub>
      <v-btn-stub
        activeclass=""
        class="hidden-md-and-down"
        tag="button"
        text="true"
        to="/contacts"
        type="button"
      >
        
        Contacts
      
      </v-btn-stub>
       
      <v-spacer-stub />
       
      <search-stub
        items=""
        value=""
      />
       
      <v-spacer-stub />
       
      <v-icon-stub
        class="hidden-md-and-down"
      >
        
        fas fa-shopping-cart fa-2x
      
      </v-icon-stub>
       
      <v-badge-stub
        bordered="true"
        bottom="true"
        class="hidden-md-and-down"
        color="yellow darken-3"
        content="6"
        label="$vuetify.badge"
        offsetx="10"
        offsety="20"
        transition="scale-rotate-transition"
        value="true"
      />
       
      <v-spacer-stub />
       
      <div
        class="text-center hidden-md-and-down"
      >
        <v-menu-stub
          closedelay="0"
          closeonclick="true"
          closeoncontentclick="true"
          contentclass=""
          maxheight="auto"
          maxwidth="auto"
          nudgebottom="0"
          nudgeleft="0"
          nudgeright="0"
          nudgetop="0"
          nudgewidth="0"
          offsety="true"
          opendelay="0"
          openonclick="true"
          origin="top left"
          transition="v-menu-transition"
        >
           
          <account-stub />
        </v-menu-stub>
      </div>
    </v-container-stub>
  </v-app-bar-stub>
   
  <v-navigation-drawer-stub
    fixed="true"
    height="100%"
    minivariantwidth="56"
    mobilebreakpoint="1264"
    src=""
    tag="aside"
    temporary="true"
    width="256"
  >
    <categories-stub
      class="col-md-12 col-12"
    />
  </v-navigation-drawer-stub>
</div>
`;
